react/src/components/StorageSelect.tsx

```diff
@@ -0,0 +1,117 @@
+import { useSuspendedBackendaiClient } from '../hooks';
+import Flex from './Flex';
+import { InfoCircleOutlined } from '@ant-design/icons';
+import { useControllableValue } from 'ahooks';
+import { Select, SelectProps, Badge, Button } from 'antd';
+import _ from 'lodash';
+import React, { useEffect, useMemo } from 'react';
+import { useTranslation } from 'react-i18next';
+import { useQuery } from 'react-query';
+
+export type VolumeInfo = {
+  id: string;
+  backend: string;
+  capabilities: string[];
+  usage: {
+    percentage: number;
+  };
+  sftp_scaling_groups: string[];
+};
+interface Props extends Omit<SelectProps, 'value' | 'onChange'> {
+  value?: string | VolumeInfo;
+  onChange?: React.Dispatch<React.SetStateAction<VolumeInfo | undefined>>;
+  autoSelectType?: 'usage' | 'default';
+  showUsageStatus?: boolean;
+}
+const StorageSelect: React.FC<Props> = ({
+  value,
+  onChange,
+  autoSelectType,
+  showUsageStatus,
+  ...selectProps
+}) => {
+  const { t } = useTranslation();
+
+  const baiClient = useSuspendedBackendaiClient();
+
+  const { data: vhostInfo, isLoading: isLoadingVhostInfo } = useQuery<{
+    default: string;
+    allowed: Array<string>;
+    volume_info: any;
+  }>('vhostInfo', () => {
+    return baiClient.vfolder.list_hosts();
+  });
+
+  const [state, setState] = useControllableValue({ value, onChange });
+
+  useEffect(() => {
+    if (!autoSelectType) return;
+
+    let selectedHost = vhostInfo?.default ?? vhostInfo?.allowed[0] ?? '';
+    if (autoSelectType === 'usage') {
+      const lowestUsageHost = _.minBy(
+        _.map(vhostInfo?.allowed, (host) => ({
+          host,
+          volume_info: vhostInfo?.volume_info[host],
+        })),
+        'volume_info.usage.percentage',
+      );
+      selectedHost = lowestUsageHost?.host || selectedHost;
+    }
+    setState({
+      id: selectedHost,
+      ...(vhostInfo?.volume_info[selectedHost] || {}),
+    });
+  }, [vhostInfo]);
+
+  const optionRender = useMemo(() => {
+    const status = ['success', 'warning', 'error'];
+    return (option: any) => {
+      const percentage =
+        vhostInfo?.volume_info[option.label]?.usage?.percentage;
+      const idx = percentage < 70 ? 0 : percentage < 90 ? 1 : 2;
+
+      if (showUsageStatus) {
+        return (
+          <Flex justify="between" align="center">
+            <Badge
+              // @ts-ignore
+              status={status[idx]}
```

Comment:
We already have color setting function in index. Please refer to this code below and update Badge component props.
https://github.com/lablup/backend.ai-webui/blob/ff29a586a1e7af517289fa1999450692f638282e/react/src/helper/index.tsx#L227-L233